From 88b87fe934158ab1a475cd0bcd80ee92c25a60da Mon Sep 17 00:00:00 2001
From: Lorenzo Pirritano <lpirritano@anaconda.com>
Date: Tue, 16 Jan 2024 21:20:22 -0600
Subject: [PATCH] bypass openssl

---
 setup.py | 23 ++---------------------
 1 file changed, 2 insertions(+), 21 deletions(-)

diff --git a/setup.py b/setup.py
index 0248577..5f14d1f 100644
--- a/setup.py
+++ b/setup.py
@@ -225,12 +225,10 @@ class BuildExtCommand(build_ext):
       module.define_macros.append(('_GNU_SOURCE', '1'))
       module.define_macros.append(('USE_MACH_PROC', '1'))
       module.extra_compile_args.append('-std=c99')
-      module.include_dirs.append('/usr/local/opt/openssl/include')
       module.include_dirs.append('/opt/local/include')
       module.library_dirs.append('/opt/local/lib')
       module.include_dirs.append('/usr/local/include')
       module.library_dirs.append('/usr/local/lib')
-      module.library_dirs.append('/usr/local/opt/openssl/lib')
     elif building_for_freebsd:
       module.define_macros.append(('_GNU_SOURCE', '1'))
       module.define_macros.append(('USE_FREEBSD_PROC', '1'))
@@ -267,25 +265,8 @@ class BuildExtCommand(build_ext):
     if self.dynamic_linking:
       module.libraries.append('yara')
     else:
-      # Is OpenSSL available?
-      if (has_function('OpenSSL_add_all_algorithms',
-                       includes=['openssl/evp.h'],
-                       include_dirs=module.include_dirs + openssl_include_dirs,
-                       libraries=module.libraries + openssl_libraries,
-                       library_dirs=module.library_dirs + openssl_library_dirs)
-          # In case OpenSSL is being linked statically
-          or has_function('OpenSSL_add_all_algorithms',
-                       includes=['openssl/evp.h'],
-                       include_dirs=module.include_dirs + openssl_include_dirs,
-                       libraries=module.libraries + openssl_libraries + ['dl', 'pthread', 'z'],
-                       library_dirs=module.library_dirs + openssl_library_dirs)
-          ):
-        module.define_macros.append(('HASH_MODULE', '1'))
-        module.define_macros.append(('HAVE_LIBCRYPTO', '1'))
-        module.libraries.extend(openssl_libraries)
-        module.include_dirs.extend(openssl_include_dirs)
-        module.library_dirs.extend(openssl_library_dirs)
-      elif building_for_windows:
+      # bypass OpenSSL
+      if building_for_windows:
         # OpenSSL is not available, but in Windows we can rely on Wincrypt.
         module.define_macros.append(('HASH_MODULE', '1'))
         module.define_macros.append(('HAVE_WINCRYPT_H', '1'))
-- 
2.39.1

